src/apps/api/tests/test_phases.py

```diff
@@ -46,3 +49,42 @@ def test_normal_user_cannot_re_run_whole_phase(self):
         resp = self.client.get(reverse('phases-rerun_submissions', kwargs={'pk': self.phase.id}))
         assert resp.status_code == 403, 'Did not raise permission denied and should have'
         assert self.phase.submissions.count() == 4
+
+
+class CompetitionPhaseMigrationValidation(TestCase):
+    def setUp(self):
+        self.user = UserFactory(username='test')
+        self.competition = CompetitionFactory(created_by=self.user, id=1)
+        self.phase = PhaseFactory(competition=self.competition)
+        self.leaderboard = LeaderboardFactory(competition=self.competition)
+
+    def serialize_and_validate_data(self):
+        serializer = CompetitionSerializer(self.competition)
+        data = serializer.data
+
+        with pytest.raises(ValidationError) as exception:
+            serializer = CompetitionSerializer(data=data)
+            serializer.is_valid(raise_exception=True)
+
+        return self, exception
```

Comment:
You can probably just return the exception here, since you aren't using the `self` anywhere

Reply:
Function now returns only exception